### PR TITLE
[FLINK-16390][sdk] Add view and clear methods to PersistedTable

### PR DESCRIPTION
**Pull request by sjwiesman**
Adds common collection methods to PersistedTable. These methods are already available on PersistedAppendingBuffer. 

Note: I have `view` return an empty iterable instead of null. This is different than the default return value for `MapState#entries` but is consistent with `PersistedAppendingBuffer#view`.

### CODE REVIEW

File: statefun-flink/statefun-flink-core/src/main/java/org/apache/flink/statefun/flink/core/state/FlinkTableAccessor.java [hunk 1]
```diff
@@ -55,4 +57,19 @@ public void remove(K key) {
       throw new RuntimeException(e);
     }
   }
+
+  @Override
+  public Iterable<Map.Entry<K, V>> view() {
+    try {
+      Iterable<Map.Entry<K, V>> entries = handle.entries();
+      return entries == null ? Collections.emptyList() : entries;
```
Comment (tzulitai) [inline statefun-flink/statefun-flink-core/src/main/java/org/apache/flink/statefun/flink/core/state/FlinkTableAccessor.java]: Actually, I think Flink user `MapState` never returns `null` for `entries()` / `keys()` / `values()`.

The actual `MapState` handle that the user gets is always a `UserFacingMapState` in Flink that wraps the actual RocksDB / heap map state handle, which uses an empty map state if the wrapped state handle returns `null` for those methods.

File: statefun-flink/statefun-flink-core/src/main/java/org/apache/flink/statefun/flink/core/state/FlinkTableAccessor.java [hunk 2]
```diff
@@ -55,4 +57,19 @@ public void remove(K key) {
       throw new RuntimeException(e);
     }
   }
+
+  @Override
+  public Iterable<Map.Entry<K, V>> view() {
```
Comment (tzulitai) [inline statefun-flink/statefun-flink-core/src/main/java/org/apache/flink/statefun/flink/core/state/FlinkTableAccessor.java]: Do you intend the returned iterable to be modifiable (i.e. the iterator supports `remove()`)?
If so, maybe we should just call this `entries()` instead.

We called the method "view" in the `PersistedAppendingBuffer` to emphasize the fact that the returned iterable is unmodifiable.

File: statefun-flink/statefun-flink-core/src/main/java/org/apache/flink/statefun/flink/core/state/FlinkTableAccessor.java (hunk 1 above)
Comment (sjwiesman) [reply statefun-flink/statefun-flink-core/src/main/java/org/apache/flink/statefun/flink/core/state/FlinkTableAccessor.java]: Looks like I have some Flink jobs I can remove some null checks from :)

**Comment by sjwiesman** [comment]
@tzulitai I've updated based on your comments. I originally went with view to be consistent with `PersistedAppendingBuffer`, but since they have different semantics I switched it back to entries.

**Comment by tzulitai** [comment]
Thanks for the update @sjwiesman.

+1 LGTM, merging ..

**Pull request closed without merging**